### PR TITLE
Home search

### PR DESCRIPTION
**Pull request by lookupdaily**
Changes to styles and text
Changed banner colour and banner font colours so that error messages will have sufficient colour contrast and removed colons from headings on trust details page table

Changes to find a trust (home and search results page)

Home page - changed search button icon to a green search button, also changed the search by text and added hint text
Added autocomplete to the search bar, with two routes - select a trust and go direct to trust details, or search for text and go to search results page
Search results page includes trust name as a link to trust details page, ID numbers and academies in trust
search results page to include number of results found and search bar to search again
Change the name from TRAMS
Changed service name to 'Find information about academies and trusts'

Have included changes on the index version page too

### CODE REVIEW

File: app/assets/sass/version-3/_components.scss [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      .autocomplete__option-hint {
+        color: govuk-colour("white");
+      }
+    }
+
+  }
+
+.govuk-button {
+  margin-left: 10px;
+}
```
Comment (dynamictulip) [inline app/assets/sass/version-3/_components.scss:37]: Should this functionality be in a new css class so it doesn't affect all `.govuk-button`s?

File: app/views/version-3/layouts/main.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    {% endblock %}
+  </div>
+</header>
+{% endblock %}
+
+{% block scripts %}
+  {{ super() }}
+  <script src="/public/javascripts/accessible-autocomplete.min.js"></script>
+  <script src="/public/javascripts/autocomplete.js"></script>
+{% endblock %}
```
Comment (dynamictulip) [inline app/views/version-3/layouts/main.html:64]: Missing end of file new line

File: app/views/version-3/not-found.html
Comment (dynamictulip) [inline app/views/version-3/not-found.html:1]: Is this file needed anymore?

Comment (dynamictulip) [reply app/views/version-3/not-found.html:1]: Sidenote for future - what should happen if someone searches for something that doesn't exist or gobbledegook?

Comment (lookupdaily) [reply app/views/version-3/not-found.html:1]: We can have a count of 0 on the search page?

File: app/views/version-3/search.html
Comment (dynamictulip) [inline app/views/version-3/search.html:1]: Needs a little whitespace tidy and blank line adding to the end

File: app/routes.js [hunk 3]
```diff
@@ -2,13 +2,63 @@
...
+
+router.get("/version-*/trust-details/:uid", function (request, response) {
+  const currentVersion = request.url.split("/")[1];
+  const uid = request.params.uid;
+  const trust = getTrustByUid(uid);
+  //response locals data will be used by next page render
+  response.locals.data.trust = trust;
+  //session data will be persisted for future pages
+  request.session.data.trust = trust;
+  response.redirect('../trust-details');
```
Comment (dynamictulip) [inline app/routes.js:59]: This is interesting, can it be used in other places rather than doing the `currentVersion` thing?

File: app/routes.js [hunk 4]
```diff
@@ -2,13 +2,63 @@
...
+    response.locals.data.searchResults = searchResults;
+    response.render(currentVersion + '/search');
+  } else {
+    response.locals.data.trusts = searchResults.slice(0, 10);
+    response.render(currentVersion + '/not-found');
+  }
+});
+
+
+router.get("/version-*/trust-details/:uid", function (request, response) {
```
Comment (dynamictulip) [inline app/routes.js:51]: Will `/version-*/` work when we get to version 10? If so then we should replace the other regex routes with this format. If not then we should reconsider its use here

File: app/routes.js [hunk 5]
```diff
@@ -2,13 +2,63 @@
 // For guidance on how to create routes see:
 // https://prototype-kit.service.gov.uk/docs/create-routes
 //
+const { get } = require('jquery');
```
Comment (dynamictulip) [inline app/routes.js:5]: I don't think this is actually being used.... can it be removed?

File: app/routes.js
Comment (dynamictulip) [inline app/routes.js:1]: This file is starting to get crowded. We should consider some refactoring

File: app/views/version-3/not-found.html
Comment (dynamictulip) [reply app/views/version-3/not-found.html:1]: Future problem though, not for this PR

Comment (jenhadfield-dfe) [reply app/views/version-3/not-found.html:1]: resolving for now as looking into this next

File: app/routes.js (hunk 4 above)
Comment (jenhadfield-dfe) [reply app/routes.js:51]: will revisit this when we refactor routes.js

File: app/routes.js (hunk 3 above)
Comment (jenhadfield-dfe) [reply app/routes.js:59]: will revisit this when we refactor routes.js

**Review by jenhadfield-dfe — APPROVED** [review]

**Review by dynamictulip — APPROVED** [review]
